ts/Core/Chart/Chart.ts

```diff
@@ -1186,8 +1186,20 @@ class Chart {
         }
 
         if (options && !elem) {
+            let text = options.text;
+
+            // If title text is unset, and not explicitly unset,
+            // default to lang option
+            if (
+                name === 'title' &&
+                chart.userOptions.title !== null &&
+                !Object.hasOwnProperty.call(options, 'text')
+            ) {
+                text = chart.options.lang.chartTitle;
+            }
```

Comment:
In general we want to avoid checking the `userOptions` as far as possible, and rely on the merged `options` being accurate. As far as I can see we can make this a little bit cheaper and at the same time avoid checking the user options:

```ts
            // If title text is unset, and not explicitly unset, default to lang
            // option
            const text = options.text ?? (
                name === 'title' && !('text' in options) ?
                    chart.options.lang.chartTitle :
                    void 0 // or null?
            );
```